integration/spark2/src/main/scala/org/apache/spark/sql/optimizer/CarbonLateDecodeRule.scala

```diff
@@ -698,11 +698,20 @@ class CarbonLateDecodeRule extends Rule[LogicalPlan] with PredicateHelper {
         u
       case p: Project if relations.nonEmpty =>
         val prExps = p.projectList.map { prExp =>
+          var isSubstringPresent = false
           prExp.transform {
-            case attr: AttributeReference =>
-              updateDataType(attr, attrMap, allAttrsNotDecode, aliasMap)
+            case subStr: Substring =>
```

Comment:
what about other UDF's like length , trunc etc. ?

Reply:
ok handled and updated test case also